tests/cpp_unit_tests/test_tap_position_optimizer.cpp

```diff
@@ -691,8 +691,8 @@ TEST_CASE("Test Tap position optimizer") {
         auto const& transformers_dataset =
             update_dataset.get_buffer_span<meta_data::update_getter_s, MockTransformer>();
         auto changed_components = std::vector<Idx2D>{};
-        main_core::update_component<MockTransformer>(state, transformers_dataset.begin(), transformers_dataset.end(),
-                                                     std::back_inserter(changed_components));
+        main_core::update::update_component<MockTransformer>(
+            state, transformers_dataset.begin(), transformers_dataset.end(), std::back_inserter(changed_components));
     };
```

Comment:
Wondering if we could make use of the ranges stuff to pass in directly here `transformers_dataset` rather than explicitly specify `begin` and `end`

Reply:
We might, but that would probably involve changing the implementation of most (if not all) `update_component` functions, which is out of scope here.

On the other hand, this may be a good thing to do "everywhere" when doing the major overhaul of the core: just use ranges stuff everywhere when possible and avoid being so explicit with iterators. Would you agree?

Reply:
Let's make that one of our todo points to see in the future if/when we can do it.

Reply:
I have added it to the discussion about possible future changes so it lives somewhere and we don't forget: https://github.com/orgs/PowerGridModel/discussions/39#discussioncomment-11332885. Closing comment now.